client/src/pages/ClientManagementPage/ClientManagementPage.js

```diff
@@ -32,10 +33,34 @@ function ClientManagementPage(props) {
   const [clientInfoList, setClientInfoList] = useState(null);
   const [isClientInfoListLoading, setIsClientInfoListLoading] = useState(false);
   const [requestError, setRequestError] = useState("");
+  const [hasPendingCancelationFilter, setHasPendingCancelationFilter] = useState("All");
+
+  function onCoveredOrderFilterChange(newVal) {
+    setHasPendingCancelationFilter(newVal);
+    onSearchClientButtonClick();
+    //FOR THE REVIEWER: this is like pressing "search" button when selecting the filter "covered/notcovered/all"
+    //might be ugly for someone, tell me in the review if I have to delete this.
+  }
 
   function onSearchClientButtonClick() {
     setIsClientInfoListLoading(true);
-    findClients(searchString)
+
+    let hasPendingCancelation;
+    switch(hasPendingCancelationFilter){
+      case("Covered"):
```

Comment:
You should replace Covered with "Pending Cancelation" or just "Pending". We decided to keep both the "not covered" status related to ["Insufficient Balance Reminder" story](https://polito-se2-21-09.myjetbrains.com/youtrack/issue/S202109SPG-8) in which the client should be notified (also before the availability confirmation) and the "pending cancelation" status related to this story in which the employee can see which clients (after the availability confirmation" have an insufficient wallet balance in order to contact them. So here you just need to take care of "pending cancelation" status I suppose